assignments/callbacks.js

```diff
@@ -0,0 +1,163 @@
+];
+
+// PROJECT RESTRICTION: You can't use map, reduce, or filter to solve these problems.  Only use native JavaScript for loops.
+
+// Example for loop:
+// arr = [1,2,3,4];
+// for (i = 0; i < arr.length; i++) {
+//     arr[i]; // 1,2,3,4
+// }
+
+// ==== Challenge 1 ====
+// The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
+const findCar = inventory.filter(item => item['id'] === 33);
+
+console.log(
+  `'Car 33 is a ${findCar[0]['car_year']} ${findCar[0]['car_make']} ${
+    findCar[0]['car_model']
+  }.`
+);
+// for (let i = 0; i < inventory.length; i++) {
+//     if (inventory[i].id === 33) {
+//         console.log(
+//             `${'Car 33 is a ' +
+//             inventory[i]['car_year'] +
+//             ' ' +
+//             inventory[i]['car_make'] +
+//             ' ' +
+//             inventory[i]['car_model'] +
+//             '.'}`
+//         );
+//     }
+// }
+
+// ==== Challenge 2 ====
+// The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
+function last() {
+  return inventory[inventory.length - 1];
+}
+const lastCar = inventory.filter(car => {
+  return car === last();
+});
+
+console.log(
+  `The last car is a ${lastCar[0]['car_make']} ${lastCar[0]['car_model']}.`
+);
+
+// for (let i = 0; i < inventory.length; i++) {
+//     if (inventory[i] === inventory[inventory.length - 1]) {
+//         console.log(
+//             'The last car is a ' + inventory[i]['car_make'],
+//             inventory[i]['car_model'] + '.'
+//         );
+//     }
+// }
+
+// // ==== Challenge 3 ====
+// // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
+const carModels = inventory.map(items => items['car_model']);
+
+// for (let i = 0; i < inventory.length; i++) {
+//     carModels.push(inventory[i]['car_model']);
+// }
+console.log(carModels.sort());
+
+// ==== Challenge 4 ====
+// The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
+let carYears = inventory.map(years => years['car_year']);
+// for (let i = 0; i < inventory.length; i++) {
+//     carYears.push(inventory[i]['car_year']);
+// }
+console.log(carYears.sort());
+
+// ==== Challenge 5 ====
+// The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
+let oldCars = inventory
+  .filter(old => old['car_year'] < 2000)
+  .map(old => old['car_year']);
```

Comment:
- I don't think you'd need map here (as you still get a length of 25 cars if you don't include it) because the filter would still give you all of the cars older than the year 2000. (Unless you wanted an array of just years)